packages/flame/test/components/component_test.dart

```diff
@@ -199,6 +199,103 @@ void main() {
         equals([Vector2(300, 200), Vector2(400, 500)]),
       );
     });
+
+    group('descendants', () {
+      testWithFlameGame(
+        'length must be 0 when it is called before being loaded',
+        (game) async {
+          final component = Component()..add(Component()..add(Component()));
+
+          expect(game.descendants().length, 0);
+          await game.ensureAdd(component);
+          expect(game.descendants().length, 3);
+        },
+      );
+
+      testWithFlameGame(
+        'length must be 3 when the number of added components is 3',
+        (game) async {
+          final component = Component()..add(Component()..add(Component()));
+          await game.ensureAdd(component);
+
+          expect(game.descendants().length, 3);
+        },
+      );
+
+      testWithFlameGame(
+        'length must be equal to the number of added components including '
+        'itself when the passed parameter includeSelf is true',
+        (game) async {
+          final component = Component()..add(Component()..add(Component()));
+          await game.ensureAdd(component);
+
+          expect(game.descendants(includeSelf: true).length, 4);
```

Comment:
how about one more assert:
```suggestion
          final descendants = game.descendants(includeSelf: true);
          expect(descendants.length, 4);
          // expect descendants contains game
```

Reply:
done